aiidalab_widgets_base/databases.py

```diff
@@ -304,50 +305,47 @@ def _observe_proxy_command(self, _=None):
             self.proxy_hostname = hostname
 
 
-class CodeDatabaseWidget(ipw.HBox):
+class ComputationalResourcesDatabase(ipw.HBox):
     """Extract the setup of a known computer from the AiiDA code registry."""
 
-    label = Unicode()
-    description = Unicode()
-    input_plugin = Unicode()
-    on_computer = Bool()
-    remote_abs_path = Unicode()
-    computer = Unicode()
-    prepend_text = Unicode()
-    append_text = Unicode()
+    input_plugin = traitlets.Unicode()
+    computer_config = traitlets.Dict()
+    computer_setup = traitlets.Dict()
+    code_config = traitlets.Dict()
 
     def __init__(self, **kwargs):
-        self.database = {}
 
+        self.database = requests.get(
+            "https://aiidateam.github.io/aiida-code-registry/database.json"
+        ).json()
```

Comment:
1. Can we store the default value as a module constant and otherwise make this a class constructor parameter?
2. We should never perform blocking requests as part of the widget construction. This will block any application that uses this widget during startup and also cause a complete failure to load in case that the resource is not available. This should be executed non-blocking and also with an explicit well-behaving failure mode.